ui_framework/components/button/button_icon/button_icon.js

```diff
@@ -0,0 +1,39 @@
+import React, {
+  PropTypes,
+} from 'react';
+
+import classNames from 'classnames';
+import keyMirror from 'keymirror';
+
+const KuiButtonIcon = props => {
+  const typeToClassNameMap = {
+    create: 'fa-plus',
+    delete: 'fa-trash',
+    previous: 'fa-chevron-left',
+    next: 'fa-chevron-right',
+    loading: 'fa-spinner fa-spin',
+  };
+
+  const iconClasses = classNames('kuiButton__icon kuiIcon', props.className, {
+    [typeToClassNameMap[props.type]]: props.type,
+  });
+
+  return (
+    <span className={iconClasses} />
+  );
+};
+
+KuiButtonIcon.propTypes = {
+  type: PropTypes.oneOf([
+    'create',
+    'delete',
+    'previous',
+    'next',
+    'loading',
+  ]),
```

Comment:
Actually, if you just export this array, we can still programmatically test a button is rendered with every type in the array, without having to switch back to the enum notation (re: my earlier comment about automating the testing so it automatically tests every type, as they are added).